src/main/kotlin/com/github/treepat/expression/TreepatExpression.kt

```diff
@@ -25,6 +25,16 @@ class TreepatExpression(val expression: VisitorFunction) {
             val astRoot: ASTNode = treepatVisitor.visit(tree)
             return TreepatExpression(createVisitorFunction(astRoot))
         }
+
+        fun createFromString(treepatString: String): TreepatExpression {
+            val lexer = TreepatLexer(CharStreams.fromString(treepatString))
+            val tokenStream = CommonTokenStream(lexer)
+            val treepatParser = TreepatParser(tokenStream)
+            val tree: ParseTree = treepatParser.treepat()
+            val treepatVisitor = TreepatVisitorImplementation()
+            val astRoot: ASTNode = treepatVisitor.visit(tree)
+            return TreepatExpression(createVisitorFunction(astRoot))
+        }
```

Comment:
There wasn't an easy way to create a `TreepatExpressionFactory` from a `String`, so I added a Factory for it.

Reply:
![](https://media.giphy.com/media/KxiRwO7tqXCTDVKobo/giphy.gif)

Reply:
Could we create a simple test for this method? So we can have some documentation in how would the client use this method.